### PR TITLE
Added a bunch more characters to translate

### PR DESCRIPTION
**Pull request by sroesler**
Hi Rick, I have added a bunch more characters that may be copied from MS Word and pasted into our app.

### CODE REVIEW

File: src/main/java/com/github/bordertech/utils/Symbols.java [hunk 1]
```diff
@@ -32,7 +38,34 @@
 			{"\u2039", "<"},
 			{"\u203A", ">"},
 			{"\u02DC", "~"},
-			{"\u00A0", " "}
+			{"\u0020", " "},
```
Comment (ricksbrown) [inline src/main/java/com/github/bordertech/utils/Symbols.java]: @sroesler I have to ask, what is the point of this one?

File: src/main/java/com/github/bordertech/utils/Symbols.java [hunk 2]
```diff
@@ -32,7 +38,34 @@
...
+			{"\u2002", " "},
+			{"\u2003", " "},
+			{"\u2004", " "},
+			{"\u2005", " "},
+			{"\u2006", " "},
+			{"\u2007", " "},
+			{"\u2008", " "},
+			{"\u2009", " "},
+			{"\u200A", " "},
+			{"\u200B", " "},
```
Comment (ricksbrown) [inline src/main/java/com/github/bordertech/utils/Symbols.java]: I think this should probably translate to nothing (an empty string) - see my comments on "Zero Width Space"

File: src/main/java/com/github/bordertech/utils/Symbols.java [hunk 3]
```diff
@@ -32,7 +38,34 @@
...
+			{"\u202F", " "},
+			{"\u205F", " "},
+			{"\u3000", " "},
+			{"\uFEFF", " "},
```
Comment (ricksbrown) [inline src/main/java/com/github/bordertech/utils/Symbols.java]: Same with this one, I think this should probably translate to nothing (an empty string) - see my comments on "Zero Width Space"

File: src/test/java/com/github/bordertech/utils/SymbolsTest.java [hunk 4]
```diff
@@ -57,4 +64,19 @@ public void testAsciifySymbolWithMarks() {
...
+	public void testAsciifyAllTheWhiteSpace() {
+		String in = "\u0020 \u00A0 \u1680 \u180E \u2000 \u2001 \u2002 \u2003 \u2004 \u2005 \u2006 \u2007 \u2008 \u2009 \u200A \u200B \u202F \u205F \u3000 \uFEFF";
+		String out = "                                       ";
+		Assert.assertEquals("Translate whitespace symbols", out, Symbols.asciifySymbol(in));
+	}
+
+	@Test
+	public void testAsciifyBracketsAndRelated() {
+		String in = "\uff08\uff09 \uff3b\uff3d \uff5b\uff5d \uff5f\uff60";
+		String out = "() [] {} ()";
```
Comment (ricksbrown) [inline src/test/java/com/github/bordertech/utils/SymbolsTest.java:85]: Nice I like this test, easy to read.

File: src/test/java/com/github/bordertech/utils/SymbolsTest.java [hunk 5]
```diff
@@ -57,4 +64,19 @@ public void testAsciifySymbolWithMarks() {
 		String out = "(C) (R) (TM)";
 		Assert.assertEquals("Translate mark symbols", out, Symbols.asciifySymbol(in));
 	}
+
+	@Test
+	public void testAsciifyAllTheWhiteSpace() {
+		String in = "\u0020 \u00A0 \u1680 \u180E \u2000 \u2001 \u2002 \u2003 \u2004 \u2005 \u2006 \u2007 \u2008 \u2009 \u200A \u200B \u202F \u205F \u3000 \uFEFF";
+		String out = "                                       ";
```
Comment (ricksbrown) [inline src/test/java/com/github/bordertech/utils/SymbolsTest.java]: Up to you but I'm wondering if this test might be easier to process visually if the `in` is an array and each character is compared to a single `" "`? No big deal, I'll leave it up to you, just food for thought I don't really mind.

**Comment by ricksbrown** [comment]
Thanks @sroesler much obliged, great stuff ✨- don't know where you came up with them all but they are exactly the kinds of problem characters this utility is designed for. I look forward to merging it once I get your thoughts on the characters I have queried.


**Thick Space**
Some of the whitespace characters below translate to a "thick" space. 
I agree with the approach you took in this PR which is to simply translate them all to one single space. 

While some formatting may be lost, meaning should not be in the scenarios this utility is designed for.
 
**Thin Space**
Similar to thick space this is mainly a question of formatting however some, for example HAIR_SPACE, is barely perceptible and *could* possibly cause confusion. I think we'll have to run the risk.

**Zero Width Space**
I believe that zero width space should probably be transformed to nothing, in other words removed from the string. Otherwise the user may find it confusing that their data has mysteriously had spaces injected into it. @sroesler What do you think?

**Characters Reviewed**
For the record, here are the characters added in this PR. Those in bold I have questioned.
TODO I should add these descriptions to all of the characters as comments in the code.

\u201F = [DOUBLE HIGH-REVERSED-9 QUOTATION MARK](https://www.fileformat.info/info/unicode/char/201f/index.htm)
\u201A = [SINGLE LOW-9 QUOTATION MARK](https://www.fileformat.info/info/unicode/char/201a/index.htm)
\u201B  = [SINGLE HIGH-REVERSED-9 QUOTATION MARK](https://www.fileformat.info/info/unicode/char/201b/index.htm)
\u2024 = [ONE DOT LEADER](https://www.fileformat.info/info/unicode/char/2024/index.htm)
\u2025 = [TWO DOT LEADER](https://www.fileformat.info/info/unicode/char/2025/index.htm)
\u2026 = [HORIZONTAL ELLIPSIS](https://www.fileformat.info/info/unicode/char/2026/index.htm)
**\u0020 = [SPACE](https://www.fileformat.info/info/unicode/char/0020/index.htm)**
\u00A0 = [NO-BREAK SPACE](https://www.fileformat.info/info/unicode/char/00A0/index.htm)
\u1680 = [OGHAM SPACE MARK](https://www.fileformat.info/info/unicode/char/1680/index.htm) - note that this one could either be a hyphen or a space. I think space is fine.
**\u180E = [MONGOLIAN VOWEL SEPARATOR](https://www.fileformat.info/info/unicode/char/180E/index.htm) - I think maybe this is zero width?**
\u2000 = [EN QUAD](https://www.fileformat.info/info/unicode/char/2000/index.htm)
\u2001 = [EM QUAD](https://www.fileformat.info/info/unicode/char/2001/index.htm)
\u2002 = [EN SPACE](https://www.fileformat.info/info/unicode/char/2002/index.htm)
\u2003 = [EM SPACE](https://www.fileformat.info/info/unicode/char/2003/index.htm)
\u2004 = [THREE-PER-EM SPACE](https://www.fileformat.info/info/unicode/char/2004/index.htm)
\u2005 = [FOUR-PER-EM SPACE](https://www.fileformat.info/info/unicode/char/2005/index.htm)
\u2006 = [SIX-PER-EM SPACE](https://www.fileformat.info/info/unicode/char/2006/index.htm)
\u2007 = [FIGURE SPACE](https://www.fileformat.info/info/unicode/char/2007/index.htm)
\u2008 = [PUNCTUATION SPACE](https://www.fileformat.info/info/unicode/char/2008/index.htm)
\u2009 = [THIN SPACE](https://www.fileformat.info/info/unicode/char/2009/index.htm)
\u200A = [HAIR SPACE](https://www.fileformat.info/info/unicode/char/200A/index.htm)
**\u200B = [ZERO WIDTH SPACE](https://www.fileformat.info/info/unicode/char/200B/index.htm)**
\u202F = [NARROW NO-BREAK SPACE](https://www.fileformat.info/info/unicode/char/202F/index.htm)
\u205F = [MEDIUM MATHEMATICAL SPACE](https://www.fileformat.info/info/unicode/char/205F/index.htm)
\u3000 = [IDEOGRAPHIC SPACE](https://www.fileformat.info/info/unicode/char/3000/index.htm)
**\uFEFF = [ZERO WIDTH NO-BREAK SPACE](https://www.fileformat.info/info/unicode/char/FEFF/index.htm)**
\uFF08 = [FULLWIDTH LEFT PARENTHESIS](https://www.fileformat.info/info/unicode/char/FF08/index.htm)
\uFF09 = [FULLWIDTH RIGHT PARENTHESIS](https://www.fileformat.info/info/unicode/char/FF09/index.htm)
\uFF3B = [FULLWIDTH LEFT SQUARE BRACKET](https://www.fileformat.info/info/unicode/char/FF3B/index.htm)
\uFF3D = [FULLWIDTH RIGHT SQUARE BRACKET](https://www.fileformat.info/info/unicode/char/FF3D/index.htm)
\uFF5B = [FULLWIDTH LEFT CURLY BRACKET](https://www.fileformat.info/info/unicode/char/FF5B/index.htm)
\uFF5D = [FULLWIDTH RIGHT CURLY BRACKET](https://www.fileformat.info/info/unicode/char/FF5D/index.htm)
\uFF5F = [FULLWIDTH LEFT WHITE PARENTHESIS](https://www.fileformat.info/info/unicode/char/FF5F/index.htm)
\uFF60 = [FULLWIDTH RIGHT WHITE PARENTHESIS](https://www.fileformat.info/info/unicode/char/FF60/index.htm)

File: src/main/java/com/github/bordertech/utils/Symbols.java [hunk 6]
```diff
@@ -32,7 +38,34 @@
 			{"\u2039", "<"},
 			{"\u203A", ">"},
 			{"\u02DC", "~"},
-			{"\u00A0", " "}
+			{"\u0020", " "},
+			{"\u00A0", " "},
+			{"\u1680", " "},
+			{"\u180E", " "},
```
Comment (ricksbrown) [inline src/main/java/com/github/bordertech/utils/Symbols.java]: Maybe zero width? See other comments on that...

File: src/main/java/com/github/bordertech/utils/Symbols.java (hunk 1 above)
Comment (sroesler) [reply src/main/java/com/github/bordertech/utils/Symbols.java]: Apologies, this is an oversight, will remove...

File: src/main/java/com/github/bordertech/utils/Symbols.java (hunk 2 above)
Comment (sroesler) [reply src/main/java/com/github/bordertech/utils/Symbols.java]: Yes, I have read your comments and I will make this change.

File: src/main/java/com/github/bordertech/utils/Symbols.java (hunk 3 above)
Comment (sroesler) [reply src/main/java/com/github/bordertech/utils/Symbols.java]: Yes, I have read your comments and I will make this change.

File: src/test/java/com/github/bordertech/utils/SymbolsTest.java (hunk 5 above)
Comment (sroesler) [reply src/test/java/com/github/bordertech/utils/SymbolsTest.java]: I will remove the in between blanks and I will split the space chars from the no-space chars, this should make it look better.

**Comment by ricksbrown** [comment]
LGTM thanks @sroesler - will merge this now 🎉

**Pull request merged**